containers/EmpSponsors.ts

```diff
@@ -0,0 +1,81 @@
+import { createContainer } from "unstated-next";
+import { useState, useEffect } from "react";
+import { BigNumber, Event } from "ethers";
+
+import Connection from "./Connection";
+import EmpContract from "./EmpContract";
+
+interface PositionState {
+  tokensOutstanding: BigNumber | null;
+  lockedCollateral: BigNumber | null;
+}
+
+interface SponsorDictionary {
+  [address: string]: PositionState;
+}
+
+interface EmpDictionary {
+  [address: string]: SponsorDictionary;
+}
```

Comment:
nit (not important at all):

Why not call it `map` instead as it is a "mapping from A to B", `SponsorMap` and `EmpMap` is shorter and simpler as well.

Reply:
Will change because Map.length < Dictionary.length, but I've never thought of dicts as different from maps fyi. Have you?

Reply:
oh yeah, they're the same. Subjectively, I just  don't like using the term `dictionary` because it's so many letters and syllables to say (even mentally) lol.